### ISSUE TITLE
apply black only for selected lines - vim

### ISSUE THREAD

**Issue by GreatBahram** [psf/black#987]
Hello there,

The problem is I just wanted to visualize some lines in vim and then run `black` just like other vim plugins. Sadly, black does not support range of lines (I think).

Thanks.

**Comment by KangOl** [comment]
You can use [black-macchiato](https://github.com/wbolster/black-macchiato) to achieve it.

**Comment by henryiii** [comment]
Would some of the changes in 19.10 help make this natively possible now (like `-c`)?

**Comment by ichard26** [comment]
While #1352 is a duplicate of this issue, that issue has more context, information, and a solution suggestion(!), so closing this in favour of that one.

**Comment by ichard26** [comment]
Ah whoops, yea this is a different issue although is blocked by #1352. My apologies @GreatBahram for 1) the wrong closure in the first place, and 2) for taking _forever_ to get back at you.
